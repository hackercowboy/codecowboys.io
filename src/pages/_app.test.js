/* eslint-disable react/jsx-no-bind */
import React from 'react';
import { shallow } from 'enzyme';
import { useRouter } from 'next/router';

import App from './_app.page';

jest.mock('next/router', () => ({ useRouter: jest.fn() }));

describe('<App/>', () => {
  it('should render in de', () => {
    useRouter.mockImplementation(() => ({ locale: 'de' }));
    function Component() {
      return <div />;
    }
    const wrapper = shallow(<App Component={Component} pageProps={{}} />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });

  it('should render in en', () => {
    useRouter.mockImplementation(() => ({ locale: 'fr' }));
    function Component() {
      return <div />;
    }
    const wrapper = shallow(<App Component={Component} pageProps={{}} />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });
});
