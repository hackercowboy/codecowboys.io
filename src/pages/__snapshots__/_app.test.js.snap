// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> should render in de 1`] = `
"<ContextProvider value={{...}}>
  <Component />
</ContextProvider>"
`;

exports[`<App/> should render in en 1`] = `
"<ContextProvider value={{...}}>
  <Component />
</ContextProvider>"
`;
